src/QsCompiler/LlvmBindings/BitcodeModule.cs

```diff
@@ -63,18 +64,101 @@ public sealed class BitcodeModule
     {
         private LLVMModuleRef moduleHandle;
 
+        private readonly Lazy<DebugInfoBuilder> LazyDiBuilder;
+
         private BitcodeModule(LLVMModuleRef handle)
         {
             this.moduleHandle = handle;
             this.Context = ContextCache.GetContextFor(handle.Context);
+            this.LazyDiBuilder = new Lazy<DebugInfoBuilder>(() => new DebugInfoBuilder(this));
         }
 
+        /// <summary>Name of the Debug Version information module flag</summary>
+        public const string DebugVersionValue = "Debug Info Version";
+
+        /// <summary>Name of the Dwarf Version module flag</summary>
+        public const string DwarfVersionValue = "Dwarf Version";
+
+        /// <summary>Version of the Debug information Metadata</summary>
+        public const UInt32 DebugMetadataVersion = 3; /* DEBUG_METADATA_VERSION (for LLVM > v3.7.0) */
+
         /// <summary>Gets a value indicating whether the module is disposed or not.</summary>
         public bool IsDisposed { get; private set; }
 
         /// <summary>Gets the <see cref="Context"/> this module belongs to.</summary>
         public Context Context { get; }
 
+        /// <summary>Gets the Metadata for module level flags</summary>
+        public IReadOnlyDictionary<string, ModuleFlag> ModuleFlags
+        {
+            get
+            {
+                ThrowIfDisposed();
+                var retVal = new Dictionary<string, ModuleFlag>();
+                (LLVMModuleFlagEntry flags, ulong len) = this.ModuleHandle.CopyModuleFlagsMetadata();
+
+                for (uint i = 0; i < len; ++i)
+                {
+                    var behavior = flags.GetFlagBehavior(i);
+                    string key = flags.GetKey(i);
+                    var metadata = LlvmMetadata.FromHandle<LlvmMetadata>(this.Context, flags.GetMetadata(i));
+                    retVal.Add(key, new ModuleFlag((ModuleFlagBehavior)behavior, key, metadata!));
+                }
+
+                flags.Dispose();
```

Comment:
It might be worth a comment here calling out the need to explicitly clean up the flags. Just thinking out loud, is there any way this could be implemented via a `using` block?

Reply:
Good point. To support a `using` block, we'd need `flags` to be `IDisposable`, but I think the class comes directly from `LLVMSharp`. Instead, how about we used a `try...finally` here to guarantee that the flags are disposed?

Reply:
Ah, right, I forgot about the `IDisposable` interface. Yes, a try-finally sounds like it would be a good fit.

Reply:
https://github.com/microsoft/qsharp-compiler/pull/1026/commits/6211ce9423675baf1e039c98c118d43e79b96367